openmc/deplete/helpers.py

```diff
@@ -306,6 +306,12 @@ def generate_tallies(self, materials, scores):
             self._rate_tally.filters = [MaterialFilter(materials)]
             self._rate_tally_means_cache = None
             if self._nuclides_direct is not None:
+                # check if any direct tally nuclides are requested that are not
+                # already loaded with the materials. Load separately if so.
+                mat_nuclides = [n for mat in materials for n in mat.nuclides]
+                extra_nuclides = set(self._nuclides_direct) - set(mat_nuclides)
```

Comment:
A set comprehension would make sense here:
```suggestion
                mat_nuclides = {n for mat in materials for n in mat.nuclides}
                extra_nuclides = set(self._nuclides_direct) - mat_nuclides
```